packages/http-client-csharp/emitter/src/lib/client-model-builder.ts

```diff
@@ -62,33 +62,53 @@ export function createModel(sdkContext: SdkContext<NetEmitterOptions>): CodeMode
     Clients: inputClients,
     Auth: processServiceAuthentication(sdkContext, sdkPackage),
   };
+
   return clientModel;
 
   function fromSdkClients(
+    sdkContext: SdkContext<NetEmitterOptions>,
     clients: SdkClientType<SdkHttpOperation>[],
     inputClients: InputClient[],
     parentClientNames: string[],
   ) {
     for (const client of clients) {
-      const inputClient = emitClient(client, parentClientNames);
+      const inputClient = fromSdkClient(sdkContext, client, parentClientNames);
       inputClients.push(inputClient);
       const subClients = client.methods
         .filter((m) => m.kind === "clientaccessor")
         .map((m) => m.response as SdkClientType<SdkHttpOperation>);
       parentClientNames.push(inputClient.Name);
-      fromSdkClients(subClients, inputClients, parentClientNames);
+      fromSdkClients(sdkContext, subClients, inputClients, parentClientNames);
       parentClientNames.pop();
     }
   }
 
-  function emitClient(client: SdkClientType<SdkHttpOperation>, parentNames: string[]): InputClient {
+  function fromSdkClient(
+    sdkContext: SdkContext<NetEmitterOptions>,
+    client: SdkClientType<SdkHttpOperation>,
+    parentNames: string[],
+  ): InputClient {
     const endpointParameter = client.initialization.properties.find(
       (p) => p.kind === "endpoint",
     ) as SdkEndpointParameter;
     const uri = getMethodUri(endpointParameter);
     const clientParameters = fromSdkEndpointParameter(endpointParameter);
+    const clientName = getClientName(client, parentNames);
+    // see if this namespace is a sub-namespace of an existing bad namespace
+    const segments = client.clientNamespace.split(".");
+    const lastSegment = segments[segments.length - 1];
+    if (lastSegment === clientName) {
+      // this segment is bad
```

Comment:
nit: instead of saying "bad" can we be more specific about what the problem is.

Reply:
I added a few more comment to explain why this is not good.